spec/index.bs

```diff
@@ -9342,7 +9397,61 @@ The determination is based on |descriptor|.{{GPURenderPipelineDescriptor/multisa
 
 ### Fragment Processing ### {#fragment-processing}
 
-Issue: fill out this section
+Fragment processing stage is a programmable stage of the render [=pipeline=] that
+evaluates the fragment data (generally, a color) to be written into render targets.
+
+This stage produce a <dfn dfn>Fragment</dfn> for each [=RasterizationPoint=]:
+<div algorithm="Fragment accessors" dfn-for=Fragment>
+    - <dfn dfn>destination</dfn> refers to [=FragmentDestination=].
+    - <dfn dfn>coverageMask</dfn> refers to multisample coverage mask (see [[#sample-masking]]).
+    - <dfn dfn>depth</dfn> refers to the depth in [=NDC=] coordinates.
+    - <dfn dfn>colors</dfn> refers to the list of color values,
+        one for each target in {{GPURenderPassDescriptor/colorAttachments}}.
+</div>
+
+<div algorithm>
+    <dfn abstract-op>process fragment</dfn>(rp, desc, state)
+
+    **Arguments:**
+    - |rp|: The [=RasterizationPoint=], produced by [[#rasterization]].
+    - |desc|: The descriptor of type {{GPUFragmentState}}.
+    - |state|: The active [=RenderState=].
+
+    **Returns:** [=Fragment=] or `null`.
+
+    1. Let |fragment| be a new [=Fragment=] object.
+    1. Set |fragment|.[=Fragment/destination=] to |rp|.[=RasterizationPoint/destination=].
+    1. Set |fragment|.[=Fragment/coverageMask=] to |rp|.[=RasterizationPoint/coverageMask=].
+    1. Set |fragment|.[=Fragment/depth=] to |rp|.[=RasterizationPoint/depth=].
+    1. If |desc| is not `null`:
+      1. Set the shader input [=builtins=]:
+          - Set the `position` builtin, if any, to
```

Comment:
Perhaps do a loop over all of the builtins of the entry point and set each value according to a `switch`?

(Is it possible for a shader to have the same builtin input multiple times?)